### DISCUSSION TITLE
Print file name and line number together in results?

### DISCUSSION THREAD

**Discussion by bobfang1992** [BurntSushi/ripgrep#1922] [Q&A] [answered]
Hi as I am using rg I get the result like this:

![01A93397-15BF-4076-B572-6A4D5E412D73](https://user-images.githubusercontent.com/44713534/124155668-a433c700-da8e-11eb-80e8-b58759f3fbac.png)

But in reality lots of times I am in a vscode terminal and if the result include the file name and line number together I can simply `Cmd + click` and it will take me to the exact location of the result. But now I need to click the file name and then go the line no. myself.

Any option to include the line number and file name together?

Thanks!

**Comment by BurntSushi** [comment, marked as answer]
Sure. Use `--no-heading`.

**Comment by sergeevabc** [comment]
Let me use this beautiful screenshot as a base for the next related question.
Here we see results grouped under filenames and each result starts with a line number.

Now I extend the chain of commands
- either with `| clip` to send output to Windows clipboard,
- or with `> output.txt` to send output to some file.

And here an unexpected turn happens:
each result gets prepended with the corresponding filename, so headings are gone, lost.

Instead of
```
myFile1.txt
  result01
  result02
```
the contents of either clipboard or output.txt become
```
myFile1.txt: result01
myFile1.txt: result02
```

Aggrrhh!

How to pass results further exactly as they are in the screenshot,
that is grouped under filenames but without filenames next to each result?

**Reply by BurntSushi** [reply]
Use `--heading --line-number`. ripgrep enables them (along with colors) automatically when printing to a tty. But when you pipe it into something else (i.e., not a tty), then ripgrep disables colors, headings and even line numbers. It "devolves" to plain grep output.

**Reply by BurntSushi** [reply]
Also would really appreciate if you opened new questions instead of necro-bumping an old one. Consider now that your question is now buried and tangled with another one. So anyone perusing the list of questions won't see yours.
